src/Browser/LoginButton.js

```diff
@@ -1,108 +1,199 @@
-const HumanInteractionInterface = require('../Auth/HumanInteractionInterface');
+const Cookies = require('./CookieUtils');
 const AuthStates = require('../Auth/AuthStates');
+const AuthController = require('../Auth/AuthController');
+const Service = require('../Service');
+const Messages = require('../Auth/LoginMessages');
+const utils = require('../utils');
 
 /**
  * @memberof Pryv.Browser
  */
-class LoginButton extends HumanInteractionInterface {
+class LoginButton {
 
-  /**
-   * @param {Browser/AuthController} auth 
-   */
-  
-  constructor (authController) {
-    super(authController);
+  constructor(authSettings, service) {
+    this.authSettings = authSettings;
+    this.service = service;
+    this.serviceInfo = service.infoSync();
   }
-   
+
   /**
    * setup button and load assets
    */
   async init () {
-    this.auth.store.stateChangeListners.push(this.onStateChange.bind(this));
-    this._setupButton();
-    await this._loadAssets();   
+    // initialize button visuals
+    setupButton(this);
+    this.languageCode = this.authSettings.authRequest.languageCode || 'en';    
+    this.messages = Messages(this.languageCode);
```

Comment:
Looks like repeated logic with authController

Reply:
haven't cleaned that yet.